nixos/modules/services/networking/go-away.nix

```diff
@@ -0,0 +1,257 @@
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    bind = lib.mkOption {
+                      default = { };
+                      type = types.submodule {
+                        freeformType = yamlFormat.type;
+                        options = {
+                          address = lib.mkOption {
+                            type = types.str;
+                            default = "${config.bindHost}:${builtins.toString config.bindPort}";
+                            readOnly = true;
+                          };
+                          tls-certificate = lib.mkOption {
+                            type = types.nullOr types.str;
+                            default =
+                              if config.tlsCertificate != null then
+                                "/run/credentials/${instanceName name}.service/tlsCertificate"
+                              else
+                                null;
+                            readOnly = true;
+                          };
+                          tls-key = lib.mkOption {
+                            type = types.nullOr types.str;
+                            default =
+                              if config.tlsKey != null then "/run/credentials/${instanceName name}.service/tlsKey" else null;
+                            readOnly = true;
+                          };
+                          # TODO
+                        };
+                      };
+                    };
+                    # TODO
+                  };
+                };
+              };
+
+              policy = lib.mkOption {
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    # TODO
+                  };
+                };
+              };
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = lib.mkIf (enabledInstances != { }) {
+    environment.etc = lib.mapAttrs' (
+      name: instance:
+      lib.nameValuePair (policyEtcPath name) ({
+        source = policyFile name instance;
+      })
+    ) enabledInstances;
+    systemd.services = lib.mapAttrs' (
+      name: instance:
+      lib.nameValuePair "${instanceName name}" {
+        description = "Go-Away (${if name == "" then "default" else name} instance)";
+        wantedBy = [ "multi-user.target" ];
+        wants = lib.lists.optional (
+          instance.useACMEHost != null
+        ) "acme-finished-${instance.useACMEHost}.target";
+        after = [
+          "network.target"
+        ] ++ (lib.lists.optional (instance.useACMEHost != null) "acme-${instance.useACMEHost}.service");
```

Comment:
Are the dependencies on ACME really necessary? The ACME service runs pretty early already and creates self-signed certificates before the final cert is acquired.

You could instead add `security.acme.certs.${instance.useACMEHost}.reloadServices = [ "go-away-${name}.service ]`, which would be required for cert renewals anyway.